src/main/java/io/anserini/util/TrainingDataGenerator.java

```diff
@@ -0,0 +1,360 @@
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.en.EnglishAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.CheckHits;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+import org.kohsuke.args4j.OptionHandlerFilter;
+import org.kohsuke.args4j.ParserProperties;
+import org.openrdf.model.Literal;
+import org.openrdf.model.ValueFactory;
+import org.openrdf.model.impl.SimpleValueFactory;
+import org.openrdf.rio.ntriples.NTriplesUtil;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Generate training data (positive examples) of a particular
+ * relationship from an indexed RDF dataset, e.g., Freebase.
+ */
+public class TrainingDataGenerator {
+
+  private static final Logger LOG = LogManager.getLogger(TrainingDataGenerator.class);
+  private final Logger TRAINING_DATA_OUTPUT_FILE_EXAMPLES;
+
+  private static final String[] supportedProperties = new String[]{
+          "birthdate"
+  };
+
+  private static final String supportedPropertiesStr = String.join(", ", supportedProperties);
+
+  // Field names of the predicates in the lucene documents
+  static final String FIELD_BIRTHDATE = "http://rdf.freebase.com/ns/people.person.date_of_birth";
+  static final String FIELD_LABEL = "http://www.w3.org/2000/01/rdf-schema#label";
+
+
+  /**
+   * The arguments that this program accepts
+   */
+  public static class Args {
+
+    // Required arguments
+    @Option(name = "-index", metaVar = "[Knoweldge base Index path]", required = true, usage = "Directory contains index files for the Knowledge base")
+    String index;
+
+    @Option(name = "-output", metaVar = "[Output file path]", required = true, usage = "Output file to write training data to")
+    String output;
+
+    @Option(name = "-property", metaVar = "[Property name]", required = true, usage = "The property to generate training data for.")
+    String property;
+  }
+
+  /**
+   * Args instance
+   */
+  Args args;
+
+  /**
+   * The directory in which the RDF dataset index is stored
+   */
+  Directory kbIndexDirectory = null;
+
+  /**
+   * Indexed knowledge base/RDF dataset to retreive
+   * the training examples from
+   */
+  IndexReader kbIndexReader = null;
+
+  /**
+   * Index searcher to search in the index
+   */
+  IndexSearcher kbIndexSearcher = null;
+
+  /**
+   * Index analyzer to search
+   */
+  Analyzer kbIndexAnalyzer;
+
+  /**
+   * Simple value factory to parse literals using Sesame library.
+   */
```

Comment:
What I meant was, can we initialize the necessary member variables such  `kbIndexDirectory`, `kbIndexReader` etc. in a constructor instead of initializing them with individual methods? something like [this](https://github.com/rosequ/Anserini/blob/master/src/main/java/io/anserini/index/IndexUtils.java#L82)